Drivers/Threshold_workout/Threshold_workout.py

```diff
@@ -1,84 +1,124 @@
-#!/usr/bin/env python3
 import sys
 import time
 import os
 from mqtt_client import MQTTClient
 from Threshold_class import ThresholdWorkout
 from dotenv import load_dotenv, set_key
 
-def perform_threshold_test(threshold_workout_object):
-    print("Current Threshold Workout: ", threshold_workout_object.get_threshold_workout())
-    print("Starting threshold test in 5 seconds...")
-    time.sleep(5)
+def perform_threshold_workout(threshold_object):
+    # Countdown
+    for i in range(5):
+        print('Starting threshold workout in', 5-i)
+        time.sleep(1)
+    print('Good Luck!!!')
     
+    # Start timer
     start_time = time.time()
-    power_usage = 0
+    
+    # Fetch data from sensor while in workout duration
     try:
-        while time.time() - start_time < (threshold_workout_object.get_duration() * 60):
+        while time.time() - start_time < (threshold_object.get_duration()*60):
             time.sleep(1)
-            if threshold_workout_object.get_power_data() is not None:
-                current_power = threshold_workout_object.get_power_data()[-1]
-                power_usage += current_power
-                print("Current power: ", current_power)
-                power_usage_per_sec = threshold_workout_object.calculate_power_usage_per_second(start_time, power_usage)
-                print("Power usage per second: ", power_usage_per_sec)
-                if current_power > threshold_workout_object.threshold_power:
-                    print("Threshold power exceeded!")
-                    resistance_required = threshold_workout_object.calculate_resistance_required(current_power)
-                    print("Resistance required to reach threshold power: ", resistance_required)
+            if threshold_object.current_power is not None:
+                print('Current power:', threshold_object.current_power)
+                print('Threshold power: ', threshold_object.get_threshold_power())
             else:
-                print("No power data received")
-
+                print('No power data received')
+                
     except KeyboardInterrupt:
-        print("Test stopped")
-        print("Count of data points given: " + str(len(threshold_workout_object.get_power_data())))
-        pass
-
+        print('Workout stopped')
+        print('Count of data points given: ' + str(len(threshold_object.power_data)))
+        mqtt_client.get_client().loop_stop()
+        exit()
 
-
-def set_workout_duration(threshold_workout_object):
-    if len(sys.argv) > 1:
-        threshold_workout_object.set_duration(int(sys.argv[1]))
-        print("Duration set to " + str(sys.argv[1]) + " minutes")
+def user_input(threshold_object):
```

Comment:
We are not allowing the user to input the metrics in the CLI, remember that we want these workouts to be called from elsewhere with metrics fed into the program from command line arguments!